### PR TITLE
Funcaptcha.js | Improved

### PR DESCRIPTION
**Pull request by pneb**
- Remove useless comments 😉
- Extract `on_click` function and move it to the top
- Extract `get_task` and `get_image` functions and move it to the top
- Move `sleep` function to the top
- Extract `main` function and move it to the top
- Call `await sleep()` instead of `await sleep(1000)`
- Use `for...in` loop instead of `for (const i in $btns)`
- Use `string` for `i` instead of `const i in $btns`
- Remove useless `parseInt`
- Use `let` instead of `const` in `for...in` loop

### CODE REVIEW

File: funcaptcha.js [hunk 1]
```diff
@@ -17,25 +28,14 @@
...
-        if (task && image) {
-            const data = {task, image, index};
-            // console.log('sending', data);
-            await Net.fetch('https://api.nopecha.com/upload', {
-                method: 'POST',
-                headers: {'Content-Type': 'application/json'},
-                body: JSON.stringify(data),
-            });
-        }
+    function sleep(delay = 1000) {
```
Comment (brunoais) [inline funcaptcha.js:31]: If you are doing this, then use sleep from `utils.js` in `Time` class

**Review by brunoais — CHANGES_REQUESTED** [review]

File: funcaptcha.js (hunk 1 above)
Comment (pneb) [reply funcaptcha.js:31]: Too lazy to initiate a function from a different file. Feel free to edit this commit if it's really bothering you

File: funcaptcha.js [hunk 2]
```diff
@@ -1,51 +1,52 @@
...
+    }
+
+    function on_game_end() {
+        is_game = false;
+    }
+
+    const get_task = () => document.querySelector('#game_children_text > h2')?.innerText;
+    const get_image = () => document.querySelector('img#game_challengeItem_image')?.src;
+
+    document.addEventListener('turbolinks:load', function() {
```
Comment (Le0Developer) [inline funcaptcha.js:43]: turbolinks?

File: funcaptcha.js [hunk 3]
```diff
@@ -1,51 +1,52 @@
...
         if (task && image) {
             const data = {task, image, index};
-            // console.log('sending', data);
-            await Net.fetch('https://api.nopecha.com/upload', {
+            const response = await Net.fetch('https://api.nopecha.com/upload', {
                 method: 'POST',
                 headers: {'Content-Type': 'application/json'},
                 body: JSON.stringify(data),
             });
+            console.log(await response.text());
```
Comment (Le0Developer) [inline funcaptcha.js:14]: removing useless console.log eh?

File: funcaptcha.js (hunk 2 above)
Comment (pneb) [reply funcaptcha.js:43]: Yep! turbolinks intercepts link clicks (all the links on the page) and makes a request to that url.
turbolinks loads the response body, and then replaces the body of the current page with the body of the response.

Comment (Le0Developer) [reply funcaptcha.js:43]: Don't think turbolinks is available

File: funcaptcha.js [hunk 4]
```diff
@@ -1,51 +1,40 @@
...
     }
+}
 
+function loop() {
+    setTimeout(() => {
+        click_verify();
+        click_btns();
+        loop();
+    }, 1000);
+}
```
Comment (brunoais) [inline funcaptcha.js:38]:
```suggestion
setInterval(() => {
    click_verify();
    click_btns();
}, 1000);
```
?

**Pull request closed without merging**